manage-data/data-store/data-streams/failure-store.md

```diff
@@ -0,0 +1,611 @@
+2025-05-05T20:49:10.899Z|sXk1opYBL1dfU_1htCAE|my-datastream  |null            |[1:45] failed to parse field [id] of type [long] in document with id 'sXk1opYBL1dfU_1htCAE'. Preview of field's value: 'invalid_text'|null           |null                |null               |null                |o.e.i.m.DocumentParsingException: [1:45] failed to parse field [id] of type [long] in document with id 'sXk1opYBL1dfU_1htCAE'. Preview of field's value: 'invalid_text'
+	at o.e.i.m.FieldMapper.rethrowAsDocumentParsingException(FieldMapper.java:241)
+	at o.e.i.m.FieldMapper.parse(FieldMapper.java:194)
+	... 19 more
+Caused by: j.l.IllegalArgumentException: For input string: "invalid_text"
+	at o.e.x.s.AbstractXContentParser.toLong(AbstractXContentParser.java:189)
+	at o.e.x.s.AbstractXContentParser.longValue(AbstractXContentParser.java:210)
+	... 26 more
+|document_parsing_exception
+```
+
+:::{note}
+Because the `document.source` field is unmapped, it is absent from the SQL results.
+:::
+::::
+:::::
+
+Failure documents have a uniform structure that is handled internally by {{es}}.
+
+`@timestamp`
+:   (`date`) The timestamp at which the document encountered a failure in {{es}}.
+
+`document`
+:   (`object`) The document at time of failure. If the document failed in an ingest pipeline, then the document will be the unprocessed version of the document as it arrived in the original indexing request. If the document failed due to a mapping issue, then the document will be as it was after any ingest pipelines were applied to it.
+    
+    `document.id`
+    :   (`keyword`) The id of the original document at the time of failure.
+    
+    `document.routing`
+    :   (`keyword`, optional) The routing of the original document at the time of failure if it was specified.
+    
+    `document.index`
+    :   (`keyword`) The index that the document was being written to when it failed.
+
+    `document.source`
+    :   (unmapped object) The body of the document. This field is unmapped and unindexed to ensure failures are indexed reliably. If you need to include fields from the document source in your queries, use [runtime fields](./failure-store.md) on the search request.
```

Comment:
> unindexed to ensure failures are indexed reliably.

I think I know what you mean but this sounds a bit confusing.

Reply:
Reworded a bit in 1591e7bb